docs/source/_ext/custom_autosummary.py

```
@@ -0,0 +1,177 @@
+
+# Custom: expand {% for <var> in <module.attr> %} into toctree table
+def _patched_autosummary_run(self):
+    """Expand {% for %} lines in directive content before building the table."""
+    expanded = _expand_for_lines(list(self.content))
+    self.content = StringList(expanded, source="<custom_autosummary expanded>")
+    return _original_autosummary_run(self)
+
+
+def monkey_patch_find_autosummary_in_lines(
+    lines: List[str],
+    module: str = None,
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)